### PR TITLE
Check if tokens are returned before attempting to create token entity

### PR DESCRIPTION
**Pull request by tnorling**
Fixes a bug where an error is thrown if an access token is not returned by the server, such as a scenario where login is called with an empty request object (we only expect id_token in the response)

### CODE REVIEW

File: lib/msal-common/src/response/ResponseHandler.ts [hunk 1]
```diff
@@ -214,19 +223,32 @@ export class ResponseHandler {
      * @param stateString 
      */
     static generateAuthenticationResult(cacheRecord: CacheRecord, idTokenObj: IdToken, fromTokenCache: boolean, stateString?: string): AuthenticationResult {
-        const responseScopes = ScopeSet.fromString(cacheRecord.accessToken.target);
+        let accessToken: string;
```
Comment (pkanher617) [inline lib/msal-common/src/response/ResponseHandler.ts:226]: set these to default values (i.e. empty string)

File: lib/msal-common/src/response/ResponseHandler.ts [hunk 2]
```diff
@@ -143,44 +143,53 @@ export class ResponseHandler {
...
 
         // IdToken
-        const cachedIdToken = IdTokenEntity.createIdTokenEntity(
-            this.homeAccountIdentifier,
-            env,
-            serverTokenResponse.id_token,
-            this.clientId,
-            idTokenObj.claims.tid
-        );
+        let cachedIdToken: IdTokenEntity;
```
Comment (pkanher617) [inline lib/msal-common/src/response/ResponseHandler.ts:146]: set to default null

File: lib/msal-common/src/response/ResponseHandler.ts [hunk 3]
```diff
@@ -143,44 +143,53 @@ export class ResponseHandler {
...
-            this.homeAccountIdentifier,
-            env,
-            serverTokenResponse.access_token,
-            this.clientId,
-            idTokenObj.claims.tid,
-            responseScopes.printScopes(),
-            tokenExpirationSeconds,
-            extendedTokenExpirationSeconds
-        );
+        let cachedAccessToken: AccessTokenEntity;
```
Comment (pkanher617) [inline lib/msal-common/src/response/ResponseHandler.ts:158]: set default null

File: lib/msal-common/src/response/ResponseHandler.ts [hunk 4]
```diff
@@ -143,44 +143,53 @@ export class ResponseHandler {
...
 
         // refreshToken
-        const cachedRefreshToken = RefreshTokenEntity.createRefreshTokenEntity(
-            this.homeAccountIdentifier,
-            env,
-            serverTokenResponse.refresh_token,
-            this.clientId,
-            serverTokenResponse.foci
-        );
+        let cachedRefreshToken: RefreshTokenEntity;
```
Comment (pkanher617) [inline lib/msal-common/src/response/ResponseHandler.ts:183]: set default null

**Comment by jasonnutter** [comment]
Agree about tests, otherwise looks good!

**Pull request merged**